hail_search/queries/base.py

```diff
@@ -360,15 +315,147 @@ def import_filtered_table(self, project_samples, num_families, **kwargs):
             self._ht = self._query_table_annotations(families_ht, self._get_table_path('annotations.ht'))
             self._ht = self._filter_annotated_table(self._ht, **kwargs)
 
-    def _filter_merged_project_hts(self, project_hts, sample_data, filtered_project_hts, filtered_comp_het_project_hts, n_partitions, **kwargs):
+    def _import_and_filter_entries_ht(self, project_samples: dict, num_families: int,  n_partitions=MAX_PARTITIONS, **kwargs):
+        """
+        Imports, prefilters, and filters one entries table (a family or project table) per sample type.
+        Returns the filtered entries table and the filtered ch table.
+        """
+        entries_hts_map: dict[str, list[tuple[hl.Table, dict]]] = {}
+        project_guid, project_sample_type_data = list(project_samples.items())[0]
+        sample_types = set(project_sample_type_data.keys())
+        if num_families == 1:
+            family_guid = list(list(project_sample_type_data.values())[0].keys())[0]
+            self._load_prefiltered_family_ht(family_guid, sample_types, entries_hts_map, project_sample_type_data, **kwargs)
+        else:
+            self._load_prefiltered_project_ht(project_guid, sample_types, entries_hts_map, project_sample_type_data, **kwargs)
+
+        # If there is only one sample type, filter the single entries table.
+        if len(entries_hts_map.keys()) == 1:
+            sample_type, entries_hts = list(entries_hts_map.items())[0]
+            ht, sorted_family_sample_data = self._add_entry_sample_families(
+                ht=entries_hts[0][0], sample_data=entries_hts[0][1], sample_type=sample_type,
+            )
+            return self._filter_entries_table(ht, sorted_family_sample_data, **kwargs)
+
+        # Project has both sample types
+        unmerged_hts, unmerged_comp_het_hts = self.filter_entries_table_both_sample_types(entries_hts_map, **kwargs)
+        merged_ht = self._merge_project_hts(unmerged_hts, n_partitions)
+        merged_comp_het_ht = self._merge_project_hts(unmerged_comp_het_hts, n_partitions)
```

Comment:
`filter_entries_table_both_sample_types` is only used twice and both times its followed by the same lines of code for merging, so it would be more readable if the merge calls where part of the `filter_entries_table_both_sample_types` helper